### PR TITLE
Show file name and size on images on hover

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/element-web/issues/18197

Hover screenshots:
image
image
image

Lightbox:
image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Show file name and size on images on hover ([\#6511](https://github.com/matrix-org/matrix-react-sdk/pull/6511)). Fixes vector-im/element-web#18197 and vector-im/element-web#18197.

Preview: https://6173114140ba111645ff8fbf--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by gsouquet — APPROVED** [review]
Looks good to me 👏

**Review by Portt56 — APPROVED** [review]

**Review by SimonBrandner — CHANGES_REQUESTED** [review]
Could a border radius please be added so that it matches the image border radius?

**Comment by turt2live** [comment]
@SimonBrandner which border radius specifically? There's already a border radius in the PR.

**Comment by SimonBrandner** [comment]
> @SimonBrandner which border radius specifically? There's already a border radius in the PR.

I meant giving the title a bottom-right radius, though it might already be there and I might be blind

**Comment by turt2live** [comment]
The bottom right is somewhat deliberately not included to avoid a trapezoid feel, though maybe the design team has thoughts :)

**Comment by robintown** [comment]
I feel like this would frequently get in the way of seeing images, especially smaller ones. Just a suggestion, but could it be something that appears on hover?

**Comment by DoM1niC** [comment]
Reply a bit buggy ??? :D 
image

**Comment by SimonBrandner** [comment]
> Reply a bit buggy ??? :D 
> image
> 

Yeah, should be hidden for replies as we already show the name on the right of the image

**Comment by ShadowJonathan** [comment]
Shouldn't this only show up when the mouse is hovering over the image?

**Comment by SimonBrandner** [comment]
> Shouldn't this only show up when the mouse is hovering over the image?

See https://github.com/matrix-org/matrix-react-sdk/pull/6511#issuecomment-890029883

**Comment by robintown** [comment]
Also fixes https://github.com/vector-im/element-web/issues/18611

**Comment by DoM1niC** [comment]
Avatar images don't to show 

```
rageshake.ts:65 TypeError: Cannot read property 'getContent' of undefined
    at O.render (ImageView.tsx:545)
    at qa (react-dom.production.min.js:187)
    at Ga (react-dom.production.min.js:186)
    at qs (react-dom.production.min.js:269)
    at Sc (react-dom.production.min.js:250)
    at yc (react-dom.production.min.js:250)
    at vc (react-dom.production.min.js:250)
    at pc (react-dom.production.min.js:243)
    at react-dom.production.min.js:123
    at t.unstable_runWithPriority (scheduler.production.min.js:18)
```

File: src/components/views/messages/MImageBody.tsx [hunk 1]
```diff
@@ -419,6 +421,18 @@ export default class MImageBody extends React.Component<IBodyProps, IState> {
             gifLabel = <p className="mx_MImageBody_gifLabel">GIF</p>;
         }
 
+        let banner;
+        const isTimeline = !this.props.tileShape
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/components/views/messages/MImageBody.tsx:425]: The FilePanel has the name + size already printed below the image

File: src/components/views/messages/MImageBody.tsx [hunk 2]
```diff
@@ -446,6 +447,21 @@ export default class MImageBody extends React.Component<IBodyProps, IState> {
             gifLabel = <p className="mx_MImageBody_gifLabel">GIF</p>;
         }
 
+        let banner: JSX.Element;
+        const isTimeline = [
+            TimelineRenderingType.Room,
+            TimelineRenderingType.Search,
+            TimelineRenderingType.Thread,
+            TimelineRenderingType.Notification,
+        ].includes(this.context.timelineRenderingType);
```
Comment (SimonBrandner) [inline src/components/views/messages/MImageBody.tsx:456]: Is this trying to make sure we don't show the banner in the file panel or something else? Perhaps a comment would help here to be a little more explicit about why are we doing this

**Review by SimonBrandner — APPROVED** [review]
Otherwise LGTM!

**Pull request merged**